pontoon/base/models.py

```diff
@@ -168,6 +169,23 @@ class UserProfile(models.Model):
     quality_checks = models.BooleanField(default=True)
     force_suggestions = models.BooleanField(default=False)
 
+    locales_order = ArrayField(
+            models.PositiveIntegerField(),
+            default=list,
+            blank=True,
+    )
+    """ Defines the order of locales displayed in locale tab."""
+
+    @property
+    def sorted_locales(self):
+        locales = Locale.objects.filter(pk__in=self.locales_order)
+        return sorted(locales, key=lambda locale: self.locales_order.index(locale.pk))
+
+    @property
+    def locales_codes(self):
+        """Return the codes of locales that contributor set in his preferences."""
+        return [l.code for l in reversed(self.sorted_locales)]
```

Comment:
Rename to simply `sorted_locales_codes` maybe? And move `reversed` logic to the client side or add a comment why this and `sorted_locales` behave differently.